crates/bevy_ecs/src/query/access.rs

```diff
@@ -2,13 +2,14 @@ use crate::storage::SparseSetIndex;
 use fixedbitset::FixedBitSet;
 use std::marker::PhantomData;
 
-/// `Access` keeps track of read and write accesses to values within a collection.
+/// Tracks read and write access to specific elements in a collection.
 ///
-/// This is used for ensuring systems are executed soundly.
+/// System initialization and execution uses these to ensure soundness.
+/// See the [`is_compatible`](Access::is_compatible) and [`get_conflicts`](Access::get_conflicts) functions.
 #[derive(Debug, Eq, PartialEq, Clone)]
 pub struct Access<T: SparseSetIndex> {
     reads_all: bool,
-    /// A combined set of T read and write accesses.
+    writes_all: bool,
     reads_and_writes: FixedBitSet,
```

Comment:
```suggestion
    reads: FixedBitSet,
```

I think this is significantly clearer. We can drop a comment that writing implies reading, or even add a debug_assert